diff --git a/lib/cjs/IModelDb.js b/lib/cjs/IModelDb.js
index 02e654c6a3b39764fb47861c1cdc94c1dced8c43..08624b86cbee00765899ad1e7b443fea454f8f6e 100644
--- a/lib/cjs/IModelDb.js
+++ b/lib/cjs/IModelDb.js
@@ -650,6 +650,9 @@ class IModelDb extends core_common_1.IModel {
     performCheckpoint() {
         if (!this.isReadonly) {
             this.saveChanges();
+            this.clearCaches();
+            this[Symbols_1._nativeDb].concurrentQueryShutdown();
+            this[Symbols_1._nativeDb].clearECDbCache();
             this[Symbols_1._nativeDb].performCheckpoint();
         }
     }
